python/pyarrow/_dataset.pyx

```diff
@@ -393,6 +393,11 @@ cdef class Dataset(_Weakrefable):
         use_threads : bool, default True
             If enabled, then maximum parallelism will be used determined by
             the number of available CPU cores.
+        use_async : bool, default False
+            If enabled, the an async scanner will be used that should offer
+            better performance with high-latency/highly-parallel filesystems
+            (e.g. S3)
+
```

Comment:
Do we want to call it out as still experimental? (Ditto for R below)

Reply:
In python the entire `pyarrow.datasets` module is still experimental.  I don't see much benefit in calling this out specifically.   Given that the API of the asynchronous scanner is the same as the synchronous scanner  I don't expect it to be any more prone to change than what was there before.